gui/widgets/qopenglmousetracker.cpp

```diff
@@ -20,6 +20,66 @@ QOpenGLMouseTracker::QOpenGLMouseTracker(QWidget *parent, float lineWidth,
       m_backgroundColor(QWidget::palette().color(QWidget::backgroundRole())),
       m_lastColour(QWidget::palette().color(QWidget::backgroundRole()).rgb()) {}
 
+std::shared_ptr<rendering::ClippingPlane>
+QOpenGLMouseTracker::BuildClippingPlane(GLfloat a, GLfloat b, GLfloat c,
+                                        GLfloat d, bool active) {
+
+  auto it = m_clippingPlanesPool.begin();
+  assert(it != m_clippingPlanesPool.end());
+
+  auto clippingPlane = *it;
+  clippingPlane->SetClipPlane(a, b, c, d);
+
+  if (active) {
+    clippingPlane->Enable();
+  } else {
+    clippingPlane->Disable();
+  }
+
+  m_clippingPlanes.insert(clippingPlane);
+  m_clippingPlanesPool.erase(clippingPlane);
+
+  update();
+
+  return clippingPlane;
+}
+
+std::shared_ptr<rendering::ClippingPlane>
+QOpenGLMouseTracker::BuildClippingPlane(const QVector3D &normal,
+                                        const QVector3D &point, bool active) {
+
+  auto it = m_clippingPlanesPool.begin();
+  assert(it != m_clippingPlanesPool.end());
+
+  auto clippingPlane = *it;
```

Comment:
again this iterator should be checked that it isn't `m_clippingPlanesPool.end()` before dereferencing it